### ISSUE TITLE
UseMemo not doing what we think

### ISSUE THREAD

**Issue by msmith-techempower** [TechEmpower/react-governor#3]
If you put a `console.log` in the memorized function, it shows we still create the hook on every state update.

I’m not sure we can get around this.

That said, given that was the reason for changing actions from a function on state to an object of functions of actions, maybe we look for another solution to actions.

**Comment by msmith-techempower** [comment]
I only have my iPhone for dev right now😅

I’m thinking of simply reverting to the function on state which returns the actions object:

```JavaScript
function actionsCreator(state) {
  return {
    increment() {
      return {
        count: state.count + 1
      };
    }
  };
}
```

This has some benefits that I want (regardless of the implementation we land on): state no longer is injected into actions, which is good because the actions no longer have the “wrong” number of arguments at call time versus their signatures (`increment()` vs `increment(state)`🤔).

The problem is that *some* `state` needs to be in scope before the actions object is built or we get errors. I cannot think of a way to handle this with magic.

**Comment by msmith-techempower** [comment]
The problem with the above is that since creating the actions is a function of state, we have to recreate the actions on every render.

We are doing this with the object version of functions right now, so it’s not really a concession, I would just like to avoid it if possible.

Put another way, if state needs to be in scope at the time the actions are declared, then they necessarily must be recreated each time state changes.

Functions, at least, have the benefit of promising that a state will be provided at the time it is invoked. Maybe it is possible to separate the creation of each action from the render cycle with `useMemo` (or similar)...

**Comment by msmith-techempower** [comment]
The more I think about this, the more I am convinced that the actions themselves must be functions of state in order for the creator to only be invoked once and not on rerenders.

So, I need to excise the creation of actions from the invocation of said actions.

**Comment by msmith-techempower** [comment]
Resolved with #5
